scripts/goldpolish-target-update-mapping.py

```diff
@@ -0,0 +1,171 @@
+        self.aln_length = int(aln_length)
+        self.quality = int(quality)
+
+    def row_to_list(self):
+        """Returns class object as a list"""
+        return [
+            self.query_name,
+            self.query_len,
+            self.query_start,
+            self.query_end,
+            self.relative_strand,
+            self.target_name,
+            self.target_length,
+            self.target_start,
+            self.target_end,
+            self.num_matches,
+            self.aln_length,
+            self.quality,
+        ]
+
+def parse_args():
+    """Reads input arguments from command line"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-g",
+        "--gaps",
+        help="extracted gaps file in fasta format",
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        "-m",
+        "--mapping",
+        help="mapping file, either verbose_mapping or paf",
+        type=str,
+        required=False,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="name of updated mapping file",
+        type=str,
+        required=True,
+        default="GoldPolish-Target_mapping_updated",
+    )
+    return parser.parse_args()
+
+def parse_description(description):
+    """parses description of fasta file and returns corresponding coordinate interval"""
+    start = int(description.split("-")[0])
+    end = int(description.split("-")[1]) + 1
+    return Interval(start, end)
+
+def populate_trees(trees, name, interval):
+    """adds intervals to the interval trees in tree_dict"""
+    name_index = name.rsplit(".", 1)[0]
+    trees[name_index].addi(interval.begin, interval.end, name)
+
+def make_interval_tree(tree_dict, args):
+    "Parses fasta sequences"
+    with btllib.SeqReader(args.gaps, btllib.SeqReaderFlag.LONG_MODE) as reader:
+        for record in reader:
+            seq_name, seq_description = record.id, record.comment
+            generic_name = seq_name.rsplit(".", 1)[0]
+            if generic_name not in tree_dict:
+                tree_dict[generic_name] = IntervalTree()
+            seq_interval = parse_description(seq_description)
+            populate_trees(tree_dict, seq_name, seq_interval)
+        return tree_dict
+
+def update_paf_file(args, tree_dict):
+    "Updates paf file based on new coordinate system"
+    with open(args.mapping, encoding="utf-8") as f_in, open(
+        args.output, "w", encoding="utf-8"
+    ) as f_out:
+        writer = csv.writer(f_out, delimiter="\t")
+        reader = csv.reader(f_in, delimiter="\t", quotechar='"')
+        # iterates through all reads in the paffile
+        for row in reader:
+            row_info = PafFileRow(*row) 
+            contig_name = row_info.target_name
+            start_pos = int(row_info.target_start)
+            end_pos = int(row_info.target_end)
+            query_start = int(row_info.query_start)
+            query_end = int(row_info.query_end)
```

Comment:
The casting to ints here - isn't it redundant because you already do that within the PafFileRow constructor?